Lib/test/test_pstats.py

```diff
@@ -36,6 +40,33 @@ def test_add(self):
         stats = pstats.Stats(stream=stream)
         stats.add(self.stats, self.stats)
 
+    def test_dump_and_load_works_correctly(self):
+        try:
+            temp_storage_new = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
```suggestion
        temp_storage_new = tempfile.NamedTemporaryFile(delete=False)
        try:
```

Reply:
Make sure the temp file is created, otherwise if an exception is raised in `NamedTemporaryFile`, you'll get an error to close and remove it.

Reply:
You know you can just commit my suggestion.

Your need to put *only* the `temp_storage_new` assignment outside of the `try` block. The purpose is to *always* clear the temp file as long as it's created. In your current code, if `dump_stats` failed, your file will not be removed.